docs/lang/articles/advanced/argument_pack.md

```diff
@@ -38,24 +38,61 @@ view_params = view_params_tmpl(
+
+- With argument packs, parameter values can be cached in the device buffer, which eliminates the need for repetitive and resource-intensive memory copying.
+  ![Copying Operations Performed With ArgPacks](https://raw.githubusercontent.com/taichi-dev/public_files/master/taichi/doc/argument_pack_memory_copying.svg)
+
+
+### Argpack Nesting
 
 :::note
+This subsection covers implementation details regarding argpack. Taichi users can safely skip reading it as it does not pertain to their usage or requirements.
+:::
+
+To enhance usability and convenience, argpacks have the ability to be nested within other argpacks. This feature allows for the organization and encapsulation of multiple argument packs within a single parent argpack. By enabling nesting, users can efficiently manage and handle complex sets of arguments, improving code clarity and maintainability.
+
+To enable nested argpacks, the parent argpack creates a pointer to the argpack buffer of the nested argpack. During code generation, a recursive process is implemented to load argpack buffers from parent argpacks to child argpacks.
+
+For example, consider that we want to load a value in `Value #3.1`. We first load Arguments Buffer, then load the pointer to argpack #1. Then we load pointer to argpack #3 inside argpack buffer #1. Finally we load value #3.1 in argpack buffer #3.
 
-While argument pack interfaces are currently supported in Taichi, the internal caching mechanism is still being developed. This feature is planned to be implemented in future versions of Taichi.
+![Load Value 3.1 in Nested ArgPacks](https://raw.githubusercontent.com/taichi-dev/public_files/master/taichi/doc/argpack_nesting_structure.svg)
 
+Value #3.1 is a struct value containing a child struct and an integer. The child struct contains an integer and a float. Consider that we'd like to load this float value. This diagram illustrates the steps to load this float. It also explains `arg_id` and `arg_depth` in detail.
+
+![Load Float Value in Value 3.1](https://raw.githubusercontent.com/taichi-dev/public_files/master/taichi/doc/argload_stmt_for_argpack_nesting.svg)
+
+
+### Buffer Values in Argpack
+
+:::note
+This subsection covers implementation details regarding argpack. Taichi users can safely skip reading it as it does not pertain to their usage or requirements.
 :::
+
+In its implementation, argpack is designed to store only constant values, such as primitive types, matrices, and dataclasses, directly in its buffer. However, for resource types that require more complex handling like Ndarrays, external arrays, sparse matrices and textures, argpack takes a different approach. Rather than storing these resources directly in the argpack buffer, argpack temporarily holds them within the Python `ArgPack` class. It then passes the pointers to these resources to the kernel by storing the pointers in the argument buffer during kernel launching.
```

Comment:
```suggestion
In its implementation, an argpack is specifically designed to store constant values like primitive types, matrices, and dataclasses directly within its buffer. However, for resource types that require more intricate handling, such as Ndarrays, external arrays, sparse matrices, and textures, argpack employs a different approach. Instead of storing these resources directly in the argpack buffer, it temporarily holds them within the Python `ArgPack` class. During kernel launching, argpack passes the pointers to these resources by storing them in the argument buffer.
```